### PR TITLE
pearcleaner 3.8.3 (new cask)

### PR DESCRIPTION
**Pull request by alienator88**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-cask+is%3Aclosed+is%3Aunmerged+&type=pullrequests) (add your cask's name to the end of the search field).
- [x] `brew audit --cask --new <cask>` worked successfully.
- [x] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

---

### CODE REVIEW

**Review by khipp — COMMENTED** [review]
Thank you for your first contribution to `homebrew-cask` and this awesome app, @alienator88! 🎉

**Comment by khipp** [comment]
Thank you, @alienator88! Could you also rebase against `origin/master` to get the CI green?

**Comment by alienator88** [comment]
Thanks @khipp!
Side question, only if you know, otherwise I can look into it more.
Is it possible to make the uninstall launchctl function dependent on if the service is actually running?
It seems like on every app upgrade via brew, it asks for sudo password to uninstall the launchctl service, but it happens even if it doesn't exist.
Some users don't seem to like having to sudo every upgrade when they aren't using that launchctl service.
Was wondering if an uninstall script could handle that instead?

File: Casks/p/pearcleaner.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  homepage "https://itsalin.com/appInfo/?id=pearcleaner"
+
+  depends_on macos: ">= :ventura"
+
+  app "Pearcleaner.app"
+
+  uninstall launchctl: "com.alienator88.PearcleanerSentinel",
+            quit:      "com.alienator88.Pearcleaner"
+
+  zap trash: "~/Library/Application Support/Pearcleaner"
```
Comment (khipp) [inline Casks/p/pearcleaner.rb:18]:
```suggestion
  zap trash: [
    "~/Library/Application Support/Pearcleaner",
    "~/Library/Caches/com.alienator88.Pearcleaner",
    "~/Library/Containers/com.alienator88.Pearcleaner",
    "~/Library/HTTPStorages/com.alienator88.Pearcleaner",
    "~/Library/Preferences/com.alienator88.Pearcleaner.plist",
  ]
```

I found some additional files that we can remove.

Comment (alienator88) [reply Casks/p/pearcleaner.rb:18]: Can we add these as well, looks like some of these other folders show up when using some of the other features in the app:
```
~/Library/Application Scripts/com.alienator88.Pearcleaner
~/Library/Application Scripts/com.alienator88.Pearcleaner.FinderOpen
~/Library/Containers/com.alienator88.Pearcleaner.FinderOpen
~/Library/Group Containers/com.alienator88.Pearcleaner
~/Library/Saved Application State/com.alienator88.Pearcleaner.savedState
```

**Comment by khipp** [comment]
@alienator88 Interesting question. I looked at the implementation, and if the `launchctl` item contains a wildcard, all matching job IDs are listed before removing them. So, we should be able to get away with using `launchctl: "com.alienator88.PearcleanerSentinel*"`.

File: Casks/p/pearcleaner.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  name "Pearcleaner"
+  desc "Utility to uninstall apps and remove leftover files from old/uninstalled apps"
+  homepage "https://itsalin.com/appInfo/?id=pearcleaner"
+
+  depends_on macos: ">= :ventura"
+
+  app "Pearcleaner.app"
+
+  uninstall launchctl: "com.alienator88.PearcleanerSentinel",
+            quit:      "com.alienator88.Pearcleaner"
```
Comment (khipp) [inline Casks/p/pearcleaner.rb:16]:
```suggestion
  uninstall launchctl:  "com.alienator88.PearcleanerSentinel*",
            quit:       "com.alienator88.Pearcleaner",
            login_item: "Pearcleaner"
```

**Review by khipp — APPROVED** [review]

**Comment by alienator88** [comment]
Thanks for all the help @khipp!

**Pull request merged**